pkg/kubectl/cmd/apply.go

```diff
@@ -182,45 +182,17 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out io.Writer, options *App
 			return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving modified configuration from:\n%v\nfor:", info), info.Source, err)
 		}
 
-		if err := info.Get(); err != nil {
-			if !errors.IsNotFound(err) {
-				return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%v\nfrom server for:", info), info.Source, err)
-			}
-			// Create the resource if it doesn't exist
-			// First, update the annotation used by kubectl apply
-			if err := kubectl.CreateApplyAnnotation(info, encoder); err != nil {
-				return cmdutil.AddSourceToErr("creating", info.Source, err)
-			}
-
-			if cmdutil.ShouldRecord(cmd, info) {
-				if err := cmdutil.RecordChangeCause(info.Object, f.Command()); err != nil {
-					return cmdutil.AddSourceToErr("creating", info.Source, err)
-				}
-			}
-
-			if !dryRun {
-				// Then create the resource and skip the three-way merge
-				if err := createAndRefresh(info); err != nil {
-					return cmdutil.AddSourceToErr("creating", info.Source, err)
-				}
-				if uid, err := info.Mapping.UID(info.Object); err != nil {
-					return err
-				} else {
-					visitedUids.Insert(string(uid))
-				}
-			}
-
-			count++
-			cmdutil.PrintSuccess(mapper, shortOutput, out, info.Mapping.Resource, info.Name, dryRun, "created")
-			return nil
-		}
-
 		if !dryRun {
 			overwrite := cmdutil.GetFlagBool(cmd, "overwrite")
 			helper := resource.NewHelper(info.Client, info.Mapping)
 			patcher := NewPatcher(encoder, decoder, info.Mapping, helper, overwrite)
 
-			patchBytes, err := patcher.patch(info.Object, modified, info.Source, info.Namespace, info.Name)
```

Comment:
against master, I'm seeing an error inside `patchSimple`, where this line is returning an internal object:
```
versionedObject, _, err := p.decoder.Decode(current, nil, nil)
```

using the info from the mapping instead let the apply unit tests pass for me:
```
versionedObject, err := api.Scheme.New(p.mapping.GroupVersionKind)
```